### DISCUSSION TITLE
8-bit to decimal 3x7 segments display

### DISCUSSION THREAD

**Discussion by yousefbinali** [hneemann/Digital#1020] [Show and tell]
Made a custom component for decimal 3x7 segments display Thanks to @hneemann for answering my question on how to create a custom display. 

It is a little bit ugly but it works!

![Screenshot_6](https://user-images.githubusercontent.com/75175135/185259316-0372ab20-65f3-4e3b-baaa-080e58dcef93.png)

**Comment by hneemann** [comment]
Nice! What are the outputs for?

**Comment by yousefbinali** [comment]
These are three BCD. I made the component first as an 8-bit to BCD encoder but I changed my mind and used the output in shape class to display the digits. You can check the code [here](https://github.com/yousefbinali/seven-by-three-display) if you like, it is a bit messy.

**Reply by prlaba** [reply]
Nice.

Does display multiplexing work in Digital?  (One set of segment signals + one 'select' signal for each display.). Can the clock run fast enough in real time in Digital to make the digits appear correctly (not sequencing through each display or randomly flashing).  I suppose real-time clock frequency depends on circuit complexity.

**Reply by yousefbinali** [reply]
I haven't tried that but I suppose as you said it depends on circuit complexity.

**Reply by xotmatrix** [reply]
I built a 4-digit multiplexed display using an embedded custom shape circuit. You plug Digital's 7-segment displays into it. The LEDs have a persistence setting to reduce flicker. It only has a resolution of four clock cycles, so it's not a snappy as a custom component.

![javaw_2022-08-17_19-26-43](https://user-images.githubusercontent.com/14209329/185453738-4b7dda14-d4e3-454d-9092-e459d851d76f.png)

You can extend that with a decoder. I made several versions using ROMs and GALs. I've never built it into the module, but there is no reason one couldn't.

![javaw_2022-08-17_19-28-25](https://user-images.githubusercontent.com/14209329/185454585-be888d5a-31a9-4cb8-b8df-9ad5ea46c1bd.png)

![javaw_2022-08-18_13-12-32](https://user-images.githubusercontent.com/14209329/185454969-05b489f3-6ace-4362-8446-245b8e88154c.png)

A couple of Twitter threads about reproducing something like this in hardware.
https://twitter.com/xotmatrix/status/1264311972960653314
https://twitter.com/xotmatrix/status/1271978331089653761

**Reply by hneemann** [reply]
Multiplexing is possible without flicker by activating an "afterglow" and even a driver transistor can be used

![Plexer3Test](https://user-images.githubusercontent.com/7190563/185623764-3d290c5a-25a5-4f02-a8b9-9261a6903120.png)
:

**Reply by prlaba** [reply]
Very cool!
